pkg/release/release.go

```diff
@@ -299,3 +299,64 @@ func GetOCIManifest(tarPath string) (*ocispec.Manifest, error) {
 	// Return the embedded v1 manifest wrapped in the container/image struct
 	return &ociman.Manifest, err
 }
+
+// CopyBinaries takes the provided `rootPath` and copies the binaries sorted by
+// their platform into the pre-defined `$PWD/bin/$PLATFORM` directories.
+func CopyBinaries(rootPath string) error {
+	platformsPath := filepath.Join(rootPath, "client")
+	platforms, err := ioutil.ReadDir(platformsPath)
+	if err != nil {
+		return errors.Wrapf(err, "retrieve platforms from %s", platformsPath)
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "get current working directory")
+	}
+
+	for _, platform := range platforms {
+		if !platform.IsDir() {
+			logrus.Warnf(
+				"Skipping platform %q because it's not a directory",
+				platform.Name(),
+			)
+			continue
+		}
+
+		logrus.Infof("Copying binaries for %s platform", platform.Name())
+
+		src := filepath.Join(
+			rootPath, "client", platform.Name(), "kubernetes/client/bin",
+		)
+		dst := filepath.Join(cwd, "bin", platform.Name())
+
+		// We assume here the "server package" is a superset of the "client
+		// package"
+		serverSrc := filepath.Join(rootPath, "server", platform.Name())
+		if util.Exists(serverSrc) {
+			logrus.Infof("Server sources found in %s, copying them", serverSrc)
+			src = filepath.Join(serverSrc, "kubernetes/server/bin")
+		}
+
+		logrus.Infof("Copying binaries from %s to %s", src, dst)
+		if err := util.CopyDirContentsLocal(src, dst); err != nil {
+			return errors.Wrapf(err,
+				"copy server binaries from %s to %s", src, dst,
+			)
+		}
+
```

Comment:
I see a bit missing here, is there a call to a `release::gcs::prepare_tarball` equivalent somewhere else in the code? 

https://github.com/kubernetes/release/blob/1b2fc8b458887834bc6fdde07bf17a3ba9f9cc87/lib/releaselib.sh#L280

Reply:
It's the file copy operation in `CopyFileLocal`. If we look at `release::gcs::prepare_tarball` then we can see that it just copies a file from A to B, there is no tarball created:

https://github.com/kubernetes/release/blob/1b2fc8b458887834bc6fdde07bf17a3ba9f9cc87/lib/releaselib.sh#L106

:shrug: